src/GafferScene/DeletePoints.cpp

```diff
@@ -51,6 +51,67 @@ using namespace IECoreScene;
 using namespace Gaffer;
 using namespace GafferScene;
 
+namespace {
+
+// Copied from Instancer.cpp - maybe should be shared somehow if it gets reused?
+struct IdData
+{
+	IdData() :
+		intElements( nullptr ), int64Elements( nullptr )
+	{
+	}
+
+	void initialize( const Primitive *primitive, const std::string &name, bool throwIfMissing = false )
+	{
+		if( const IntVectorData *intData = primitive->variableData<IntVectorData>( name ) )
+		{
+			intElements = &intData->readable();
+		}
+		else if( const Int64VectorData *int64Data = primitive->variableData<Int64VectorData>( name ) )
+		{
+			int64Elements = &int64Data->readable();
+		}
+		else if( throwIfMissing )
+		{
+			throw IECore::Exception( fmt::format( "DeletePoints : No primitive variable \"{}\" found of type IntVectorData or type Int64VectorData", name ) );
+		}
+	}
+
+	size_t size() const
+	{
+		if( intElements )
+		{
+			return intElements->size();
+		}
+		else if( int64Elements )
+		{
+			return int64Elements->size();
+		}
+		else
+		{
+			return 0;
+		}
+	}
+
+	int64_t element( int i ) const
```

Comment:
Hmm, good point about consistency. I feel conflicted because there are currently lots of places in the code where we fail on PointsPrimitives with more than 2^31 points. I think consistently using unsigned int at least would make the most sense to me - it would get us twice as many points for free. But if we're not currently changing all the other places, should I make size() return `int` for consistency? Or do I just make this take a size_t? I guess it's most consistent if we consider the IdData struct independently from how we actually use it?

Reply:
> I feel conflicted because there are currently lots of places in the code where we fail on PointsPrimitives with more than 2^31 points.

Where are these places? `PointsPrimitive::m_numPoints` is `size_t`, and the size of a PrimitiveVariable is `size_t`, so `size_t` is clearly the right choice here. Are you getting mixed up with MeshPrimitive indices perhaps?